src/OpenFlight/Users/Infrastructure/MysqlUserRepository.php

```diff
@@ -29,4 +31,20 @@ public function Save(User $user): void
         $statement->execute();
     }
 
+    public function findByUsername(string $username): User
+    {
+        $sql = 'SELECT * FROM user WHERE username = :username LIMIT 1';
+        $statement = $this->mysql->PDO()->prepare($sql);
+        $statement->bindValue(':username', $username);
+        $statement->execute();
+        $userArr = $statement->fetchAll();
+
+        if (empty($userArr)) {
+            throw new IncorrectUserName();
+        }
+
+        $user = $userArr[0];
+        $uuid = new Uuid($user["Id"]);
+        return User::LoginUser($uuid, $user["Username"], $user["Name"], $user["LastName"], $user["Password"]);
```

Comment:
No hace falta crear un `NamedConstructor`. Sabemos que lo que esta guardado en la base de datos fue persistido de manera correcta, por lo tanto no hace falta validar como lo haciamos en el register (donde los datos vienen desde un cliente). 
```suggestion
        return new User($uuid, $user["Username"], $user["Name"], $user["LastName"], $user["Password"]);
```